1-bootstrap/main.tf

```diff
@@ -17,34 +17,51 @@
 locals {
   cb_config = {
     "multitenant" = {
-      repo_name    = "eab-multitenant",
       bucket_infix = "mt"
       roles = [
         "roles/container.admin"
       ]
     }
     "applicationfactory" = {
-      repo_name    = "eab-applicationfactory",
       bucket_infix = "af"
       roles        = ["roles/resourcemanager.projectIamAdmin"]
     }
     "fleetscope" = {
-      repo_name    = "eab-fleetscope",
       bucket_infix = "fs"
       roles        = []
     }
   }
+  use_csr                    = var.cloudbuildv2_repository_config.repo_type == "CSR"
+  csr_repos                  = local.use_csr ? { for k, v in var.cloudbuildv2_repository_config.repositories : k => v.repository_name } : {}
   cb_service_accounts_emails = { for k, v in module.tf_cloudbuild_workspace : k => reverse(split("/", v.cloudbuild_sa))[0] }
 }
 
 resource "google_sourcerepo_repository" "gcp_repo" {
-  for_each = local.cb_config
+  for_each = local.csr_repos
 
   project                      = var.project_id
-  name                         = each.value.repo_name
+  name                         = each.value
   create_ignore_already_exists = true
 }
 
+module "cloudbuild_repositories" {
+  count = local.use_csr ? 0 : 1
+  # TODO: update git URL to registry format after release is available
+  source = "git::https://github.com/terraform-google-modules/terraform-google-bootstrap.git//modules/cloudbuild_repo_connection?ref=a6072e0db92d976f1535dab767ad6b4331bcb4ef"
```

Comment:
I would recommend this PR wait to merge till after this upstream release is ready.

To expedite, I just reviewed this PR: https://github.com/terraform-google-modules/terraform-google-bootstrap/pull/328

Reply:
Updated `terraform-google-bootstrap` to v10